sdk/program/src/stake/instruction.rs

```diff
@@ -167,6 +167,61 @@ pub enum StakeInstruction {
     ///   3. Optional: [SIGNER] Lockup authority, if updating StakeAuthorize::Withdrawer before
     ///      lockup expiration
     AuthorizeWithSeed(AuthorizeWithSeedArgs),
+
+    /// Initialize a stake with authorization information
+    ///
+    /// This instruction is similar to `Initialize` except that the withdraw authority
+    /// must be a signer, and no lockup is applied to the account.
+    ///
+    /// # Account references
+    ///   0. [WRITE] Uninitialized stake account
+    ///   1. [] Rent sysvar
+    ///   2. [] The stake authority
+    ///   3. [SIGNER] The withdraw authority
+    ///
+    InitializeChecked,
+
+    /// Authorize a key to manage stake or withdrawal
+    ///
+    /// This instruction behaves like `Authorize` with the additional requirement that the new
+    /// stake or withdraw authority must also be a signer.
+    ///
+    /// # Account references
+    ///   0. [WRITE] Stake account to be updated
+    ///   1. [] Clock sysvar
+    ///   2. [SIGNER] The stake or withdraw authority
+    ///   3. [SIGNER] The new stake or withdraw authority
+    ///   4. Optional: [SIGNER] Lockup authority, if updating StakeAuthorize::Withdrawer before
+    ///      lockup expiration
+    AuthorizeChecked(StakeAuthorize),
+
+    /// Authorize a key to manage stake or withdrawal with a derived key
+    ///
+    /// This instruction behaves like `AuthorizeWithSeed` with the additional requirement that
+    /// the new stake or withdraw authority must also be a signer.
+    ///
+    /// # Account references
+    ///   0. [WRITE] Stake account to be updated
+    ///   1. [SIGNER] Base key of stake or withdraw authority
+    ///   2. [] Clock sysvar
+    ///   3. [SIGNER] The new stake or withdraw authority
+    ///   4. Optional: [SIGNER] Lockup authority, if updating StakeAuthorize::Withdrawer before
+    ///      lockup expiration
+    AuthorizeCheckedWithSeed(AuthorizeCheckedWithSeedArgs),
+
+    /// Set stake lockup
+    ///
+    /// This instruction behaves like `SetLockup` with the additional requirement that
+    /// the new lockup authority also be a signer.
+    ///
+    /// If a lockup is not active, the withdraw authority may set a new lockup
+    /// If a lockup is active, the lockup custodian may update the lockup parameters
+    ///
+    /// # Account references
+    ///   0. [WRITE] Initialized stake account
+    ///   1. [SIGNER] Lockup authority or withdraw authority
+    ///   2. Optional: [SIGNER] New lockup authority
```

Comment:
```suggestion
    ///   2. [SIGNER] New lockup authority
```
No longer optional, correct? (I'm about to push `transaction-status` module updates)

Reply:
Oh I see, nevermind